src/Microsoft.AspNetCore.Server.Kestrel/KestrelServerOptions.cs

```diff
@@ -57,13 +60,35 @@ public class KestrelServerOptions
             {
                 if (value.HasValue && value.Value <= 0)
                 {
-                    throw new ArgumentOutOfRangeException("value", "Value must be null or a positive integer.");
+                    throw new ArgumentOutOfRangeException(nameof(value), "Value must be null or a positive integer.");
                 }
                 _maxRequestBufferSize = value;
             }
         }
 
         /// <summary>
+        /// Gets or sets the maximum allowed size for the HTTP request line.
+        /// </summary>
+        /// <remarks>
+        /// Defaults to 8,192 bytes (8 KB).
+        /// </remarks>
+        public int MaxRequestLineSize
```

Comment:
I think we should move these settings to a sub object so the top level class isn;t polluted.